.github/workflows/ci.yaml

```diff
@@ -23,4 +23,26 @@ jobs:
       - name: Install dependencies
         run: python -m pip install tox
       - name: Run tests
-        run: tox -e unit
\ No newline at end of file
+        run: tox -e unit
+
+  integration-test:
+    name: Integration tests (microk8s)
+    timeout-minutes: 120
+    needs:
+      - lint
+      - unit-test
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v2
+      - name: Setup operator environment
+        id: operator
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          # This is needed until https://bugs.launchpad.net/juju/+bug/1977582 is fixed
+          bootstrap-options: "--agent-version 2.9.29"
```

Comment:
JFYI, you are affected there if you are testing scale-down ONLY.
AFAIK you are not performing scale down. Maybe you can use the latest 2.9.38 here?
JFMI, Have you ever tried it?